org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -105,6 +107,17 @@
 	 */
 	public static final String MEMBER_SORT_ORDER = "java.memberSortOrder"; //$NON-NLS-1$
 
+	/**
+	 * Preference key for the id(s) of the preferred content provider(s).
+	 */
+	public static final String PREFERRED_CONTENT_PROVIDER_KEY = "java.contentProvider.preferred";
+
+	/**
+	 * Preference key for the maximum size of the content cache.
+	 */
+	public static final String CONTENT_PROVIDER_CACHE_MAX_SIZE_KEY = "java.contentProvider.cache.maxSize";
```

Comment:
Does this need to be a user configurable preference? I think we should just remove it as a preference and have a constant somewhere for it.

Reply:
I'll make that change.